deepchem/feat/protein_tokenizer.py

```diff
@@ -0,0 +1,155 @@
+from deepchem.feat.base_classes import Featurizer
+from typing import Iterable
+import numpy as np
+import logging
+logger = logging.getLogger(__name__)
+
+# Integer tokens for every amino acid code in the FASTA format.
+FASTA_tokens = {
+    "A": 0,
+    "B": 1,
+    "C": 2,
+    "D": 3,
+    "E": 4,
+    "F": 5,
+    "G": 6,
+    "H": 7,
+    "I": 8,
+    "J": 9,
+    "K": 10,
+    "L": 11,
+    "M": 12,
+    "N": 13,
+    "O": 14,
+    "P": 15,
+    "Q": 16,
+    "R": 17,
+    "S": 18,
+    "T": 19,
+    "U": 20,
+    "V": 21,
+    "W": 22,
+    "Y": 23,
+    "Z": 24,
+    "X": 25,
+    "*": 26,
+    "-": 27
+}
+
+
+class ProteinTokenizer(Featurizer):
```

Comment:
We should coordinate the API style in this PR and https://github.com/deepchem/deepchem/pull/2523 since both are introducing HuggingFace tokenizers

CC @walid0925

Reply:
This PR actually wasn't modeled after existing HuggingFace tokenizers per our discussion on Thursday.

Reply:
Hmm this is definitely something to talk about! If we want to pursue HF models and upload these to the model hub this would likely need to also subclass a HF tokenizer of some type (otherwise saving tokenizer files in the right format will be difficult)